tests/test_matrix_utils.py

```
@@ -3559,9 +3564,231 @@ def fake_async_client(*_args, **_kwargs):
+        Simulate a sync operation that increments a shared call counter and fails with controlled exceptions.
+
+        Increments call_count["count"] each invocation. On the first invocation raises jsonschema.exceptions.ValidationError with message "Invalid schema"; on every subsequent invocation raises NioLocalTransportError("retry failed"). Positional and keyword arguments are ignored.
+        """
+        call_count["count"] += 1
+        if call_count["count"] == 1:
+            raise jsonschema.exceptions.ValidationError(
+                message="Invalid schema",
+                path=(),
+                schema_path=(),
+            )
+        raise NioLocalTransportError("retry failed")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)